snappy/services.go

```diff
@@ -295,20 +241,29 @@ WantedBy={{.ServiceSystemdTarget}}
 		EnvVars              string
 		SocketFileName       string
 		Restart              string
-		Type                 string
+		StopTimeout          time.Duration
+		// For snapenv.GetBasicSnapEnvVars
+		Version  string
+		SnapName string
+		Revision int
+		SnapPath string
 	}{
-		*desc,
-		filepath.Join(desc.SnapPath, desc.Start),
-		filepath.Join(desc.SnapPath, desc.Stop),
-		filepath.Join(desc.SnapPath, desc.PostStop),
+		appInfo,
+		filepath.Join(realBaseDir, appInfo.Command),
+		filepath.Join(realBaseDir, appInfo.Stop),
+		filepath.Join(realBaseDir, appInfo.PostStop),
 		systemd.ServicesTarget,
 		arch.UbuntuArchitecture(),
 		// systemd runs as PID 1 so %h will not work.
 		"/root",
 		"",
-		desc.SocketFileName,
+		socketFileName,
 		restartCond,
-		desc.Type,
+		serviceStopTimeout(appInfo),
+		appInfo.Snap.Version,
```

Comment:
Isn't this actually the same refactoring being done in this branch?  I mean, this is just about using {{.App.Snap.Version}} in that template instead of {{Version}}, isn't it?